src/playground-project.ts

```diff
@@ -838,3 +863,29 @@ const outdent = (str: string): string => {
   }
   return str.replace(RegExp(`^\\s{${shortestIndent ?? 0}}`, 'gm'), '');
 };
+
+/**
+ * Test whether two lists of Playground files are deeply equal.
+ */
+const playgroundFilesDeepEqual = (
+  filesA: SampleFile[],
+  filesB: SampleFile[]
+): boolean => {
+  if (filesA.length !== filesB.length) {
+    return false;
+  }
+  for (let i = 0; i < filesA.length; i++) {
+    const fileA = filesA[i];
+    const fileB = filesB[i];
+    if (
+      fileA.name !== fileB.name ||
+      fileA.content !== fileB.content ||
+      fileA.contentType !== fileB.contentType ||
+      fileA.hidden !== fileB.hidden ||
+      fileA.label !== fileB.label
```

Comment:
Nit: For greatest chance of short circuiting after minimal work, we could check the `content` last (which I'm assuming will have the most string comparing).

Could also do 2 loops. The first loop checks all the metadata that is relatively small ignoring any content checking. This gives a very quick way to find inequality without content. Then do a second loop over the content. 

Computers are pretty fast so no changes required.